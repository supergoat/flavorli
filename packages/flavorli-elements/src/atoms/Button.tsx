import {styled} from '@flavorli/elements';

const Button = styled.button`
  font-size: 16px;
  padding: 12px;
  border-radius: 4px;
  border: none;
  box-shadow: 1px 1px 4px #bbb;
`;

export default Button;
