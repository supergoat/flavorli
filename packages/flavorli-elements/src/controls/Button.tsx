import styled from 'styled-components';
import {IColor} from '../theme/colors';

const DEFAULT_WIDTH = '120px';
const DEFAULT_HEIGHT = '35px';

export interface IButtonProps {
  intent?: 'secondary' | 'text';
  height?: string;
  width?: string;
  color?: IColor;
}

export const Button = styled.button<IButtonProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
  height: ${p =>
    p.height ? p.height : p.intent === 'text' ? 'auto' : DEFAULT_HEIGHT};
  width: ${p =>
    p.width ? p.width : p.intent === 'text' ? 'auto' : DEFAULT_WIDTH};
  background-color: ${p =>
    p.intent === 'secondary'
      ? p.theme.colors.white
      : p.intent === 'text'
      ? 'transparent'
      : p.theme.colors.primary};
  font-weight: ${p => (p.intent === 'text' ? 'bold' : 'regular')};
  color: ${p =>
    p.color
      ? p.theme.colors[p.color]
      : p.intent === 'secondary' || p.intent === 'text'
      ? p.theme.colors.primary
      : p.theme.colors.textOnPrimary};
  border-radius: ${p => `${p.theme.spacings['4']}px`};
  border: ${p =>
    p.intent === 'text' ? 'none' : `1px solid ${p.theme.colors.primary}`};
  box-shadow: ${p =>
    p.intent === 'secondary' || p.intent === 'text'
      ? 'none'
      : p.theme.shadows.LIGHT};
  /* transition: 0.1s; */

  &:hover {
    background-color: ${p =>
      p.intent === 'secondary'
        ? p.theme.colors.secondarySurface
        : p.intent === 'text'
        ? 'none'
        : p.theme.colors.primaryDark};
  }
`;
