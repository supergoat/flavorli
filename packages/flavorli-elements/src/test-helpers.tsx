import React from 'react';
import {ThemeProvider} from 'styled-components';
import {render as testRender} from '@testing-library/react';
import {theme} from '.';

export function render(ui: React.ReactNode) {
  const utils = testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

  return utils;
}
