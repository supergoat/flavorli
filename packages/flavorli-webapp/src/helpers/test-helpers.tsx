import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import {ThemeProvider} from 'styled-components';
import {render as testRender} from '@testing-library/react';
import {theme} from '@flavorli/elements';

export function render(ui: React.ReactNode) {
  const utils = testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

  return utils;
}

export function renderWithRouter(ui: React.ReactElement, route: string = '/') {
  const history = createMemoryHistory({initialEntries: [route]});
  const Wrapper = ({children}: {children: React.ReactNode}) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, {
      wrapper: Wrapper as React.ComponentType,
    }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
}
