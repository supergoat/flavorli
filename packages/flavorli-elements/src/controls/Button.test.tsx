import React from 'react';
import {render} from '../test-helpers';
import {axe} from 'jest-axe';
import {Button} from './Button';

it('should not have any axe violations', async () => {
  const {container} = render(<Button>Hello World</Button>);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});

it('should have display the default button', () => {
  const {container} = render(<Button>Hello World</Button>);
  expect(container).toMatchSnapshot();
});
