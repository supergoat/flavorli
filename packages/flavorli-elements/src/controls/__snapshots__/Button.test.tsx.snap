// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have display the default button 1`] = `
<button
  class="sc-AykKC cPGfCk"
>
  Hello World
</button>
`;

exports[`should have the correct styles 1`] = `
<button
  class="sc-AykKC cPGfCk"
>
  Hello World
</button>
`;
